lib/src/exceptions/posthog_error_tracking_autocapture_integration.dart

```diff
@@ -0,0 +1,175 @@
+
+/// Handles automatic capture of Flutter and Dart exceptions
+class PostHogErrorTrackingAutoCaptureIntegration {
+  final PostHogErrorTrackingConfig _config;
+  final PosthogFlutterPlatformInterface _posthog;
+
+  // Store original handlers (we'll chain with them from our handler)
+  FlutterExceptionHandler? _originalFlutterErrorHandler;
+  ErrorCallback? _originalPlatformErrorHandler;
+
+  bool _isEnabled = false;
+
+  static PostHogErrorTrackingAutoCaptureIntegration? _instance;
+
+  PostHogErrorTrackingAutoCaptureIntegration._({
+    required PostHogErrorTrackingConfig config,
+    required PosthogFlutterPlatformInterface posthog,
+  })  : _config = config,
+        _posthog = posthog;
+
+  /// Install the autocapture integration (can only be installed once)
+  static PostHogErrorTrackingAutoCaptureIntegration? install({
+    required PostHogErrorTrackingConfig config,
+    required PosthogFlutterPlatformInterface posthog,
+  }) {
+    if (_instance != null) {
+      debugPrint(
+          'PostHog: Error tracking autocapture integration is already installed. Call PostHogErrorTrackingAutoCaptureIntegration.uninstall() first.');
+      return null;
+    }
+
+    _instance = PostHogErrorTrackingAutoCaptureIntegration._(
+      config: config,
+      posthog: posthog,
+    );
+
+    if (config.captureUnhandledExceptions) {
+      _instance!.start();
+    }
+
+    return _instance;
+  }
+
+  /// Uninstall the autocapture integration
+  static void uninstall() {
+    if (_instance != null) {
+      _instance!.stop();
+      _instance = null;
+    }
+  }
+
+  /// Start automatic exception capture
+  void start() {
+    if (_isEnabled) return;
+
+    _isEnabled = true;
+
+    // Set up Flutter error handler if enabled
+    if (_config.captureFlutterErrors) {
+      _setupFlutterErrorHandler();
+    }
+
+    // Set up platform error handler if enabled
+    if (_config.captureDartErrors) {
+      _setupPlatformErrorHandler();
+    }
+  }
+
+  /// Stop automatic exception capture (restores original handlers)
+  void stop() {
+    if (!_isEnabled) return;
+
+    _isEnabled = false;
+
+    // Restore original handlers
+    if (_originalFlutterErrorHandler != null) {
+      FlutterError.onError = _originalFlutterErrorHandler;
+      _originalFlutterErrorHandler = null;
+    }
+
+    if (_originalPlatformErrorHandler != null) {
+      PlatformDispatcher.instance.onError = _originalPlatformErrorHandler;
+      _originalPlatformErrorHandler = null;
+    }
+  }
+
+  /// Flutter framework error handler
+  void _setupFlutterErrorHandler() {
+    // prevent circular calls
+    if (FlutterError.onError == _posthogFlutterErrorHandler) {
+      return;
+    }
+
+    _originalFlutterErrorHandler = FlutterError.onError;
+
+    FlutterError.onError = _posthogFlutterErrorHandler;
+  }
+
+  void _posthogFlutterErrorHandler(FlutterErrorDetails details) {
+    // don't capture silent errors (could maybe be a config?)
+    if (!details.silent) {
+      _captureException(
```

Comment:
a few more details we should capture eg
https://github.com/getsentry/sentry-dart/blob/a69a51fd1695dd93024be80a50ad05dd990b2b82/packages/flutter/lib/src/integrations/flutter_error_integration.dart#L35-L60